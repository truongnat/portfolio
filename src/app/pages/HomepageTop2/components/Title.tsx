import styled, { keyframes } from 'styled-components/macro';
import { speed } from '../../../../constants/common';
import { media } from '../../../../styles/media';

const scale = keyframes`
   0% {
        transform: scale(1);
    }
    100% {
        transform: scale(1.02);
    }
`;
export const Title = styled.h1`
  font-size: 2rem;
  font-weight: bold;
  letter-spacing: 2rem;
  margin: 0;
  padding: 1rem 0;
  color: #fff;
  animation: ${scale} ${speed * 1.25}s ease-in-out infinite alternate;
  transform-origin: center;
  text-transform: uppercase;
  ${media.small`
		font-size : 1.25rem;
		letter-spacing: 0.5rem;
  `};
  ${media.xSmall`
		font-size: 1.5rem;
		letter-spacing: 0.75rem;
	`};
  ${media.medium`
        font-size : 2rem;
        letter-spacing: 1rem;
    `}
`;
