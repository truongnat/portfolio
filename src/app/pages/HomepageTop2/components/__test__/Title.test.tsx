import * as React from 'react';

import { createRenderer } from 'react-test-renderer/shallow';

import { Title } from '../Title';

const shallowRenderer = createRenderer();

describe('<Title />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<Title />);
    const renderOutput = shallowRenderer.getRenderOutput();
    expect(renderOutput).toMatchSnapshot();
  });
});
