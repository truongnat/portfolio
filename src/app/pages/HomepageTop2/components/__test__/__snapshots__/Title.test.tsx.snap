// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render and match the snapshot 1`] = `
.c0 {
 font-size: 2rem;
  font-weight: bold;
  letter-spacing: 2rem;
  margin: 0;
  padding: 1rem 0;
  color: #fff;
  animation: scale 1.875s ease-in-out infinite alternate;
  transform-origin: center;
  text-transform: uppercase;
}

@keyframes scale {
	  0% {
        transform: scale(1);
    }
    100% {
        transform: scale(1.02);
    }
}

@media only screen and (min-width: 320px) {
  .c0 {
   font-size : 1.25rem;
	 letter-spacing: 0.5rem;
  }
}

@media only screen and (min-width: 600px) {
  .c0 {
  	font-size: 1.5rem;
		letter-spacing: 0.75rem;
  }
}

@media only screen and (min-width: 1024px) {
  .c0 {
		font-size : 2rem;
		letter-spacing: 1rem;
  }
}

<h1
  class="c0"
/>
`;
